### PR TITLE
Updating documentation on Custom Trees

### PR DESCRIPTION
**Pull request by marcemarc**
I started looking at updating the config page for Custom Trees, as was my strategy, however found very odd that we don't have anything on Creating Trees? or maybe it is somewhere else, anyway here is the beginning of trying to document trees as per issue #219 -  but figured this initial very barebones introduction to creating a custom tree in code - is  better than the '……coming soon' message!

So thought maybe we could merge this in first as a quick win? , first step forwards?

Then there is more to come on this,to provide the trees.config documentation separately from the introduction to the subject and the example could be fleshed out further - I'm just saying I'm not done with this, and we're probably going to have to move some stuff around, but that's easier to do when you can see how much stuff there is,  but this is possibly worth adding now... if people agree?

### CODE REVIEW

**Comment by hemraker** [comment]
This is awesome @marcemarc 😍

File: Extending/Section-Trees/trees-v7.md [hunk 1]
```diff
@@ -6,12 +6,127 @@ This section describes how to work with and create trees with the v7 APIs.
...
+![Favourite Things Custom Tree](images/favourite-things-custom-tree.png)
+
+## Responding to Tree Actions
+
+The actions on items in an Umbraco Tree are wired 'by convention' to the loading of an AngularJS view with a corresponding name to the action, from a folder in the views section for the customTreeAlias.
+
+For example, 'Clicking on' one of our Favourite Things in our custom tree example will 'by convention' trigger the loading of an 'edit.html' view from the folder /views/favouriteThings/edit.html
+and the 'Delete' menu item we added to all our favourite things in the tree, will when selected load a view called 'delete.html' in the location /views/favouriteThings/delete.html
+
+It's recommended, particuarly if you shipping your custom tree as a package, to change the location of this default folder to the app_plugins folder and you achieve this by decorating you mvc *TreeController* with the PluginController attribute.
```
Comment (Jeavon) [inline Extending/Section-Trees/trees-v7.md]: " if you're shipping your custom tree"

File: Extending/Section-Trees/trees-v7.md [hunk 2]
```diff
@@ -6,12 +6,127 @@ This section describes how to work with and create trees with the v7 APIs.
...
+
+    {
+     "javascript": [
+       "~/App_Plugins/favouriteThings/favouriteThings.resource.js",
+       "~/App_Plugins/favouriteThings/backoffice/favouriteThingsAlias/edit.controller.js",
+       "~/App_Plugins/favouriteThings/backoffice/favouriteThingsAlias/delete.controller.js"
+     ]
+    }
+
+to load any additional javascript resources we would like to use with our AngularJS Views, and we can update our views to use these controllers with the ng-controller attribute. for example our delete view may look like this:
```
Comment (Jeavon) [inline Extending/Section-Trees/trees-v7.md]: Mixing up "your", "we" and "our", perhaps rephrase this one?

**Comment by marcemarc** [comment]
@Jeavon you our we, hopefully a little clearer now :-)

**Comment by Jeavon** [comment]
@marcemarc good to be merged unless you want a second review?

**Comment by marcemarc** [comment]
@Jeavon merged!, (@hemraker has glanced at it too) but I do intend to move this stuff around a bit too, to hopefully make this hang together better, but for now this is better than what is there.

**Comment by Jeavon** [comment]
:fireworks:

**Comment by hemraker** [comment]
⭐️ ⭐️ ⭐️ ⭐️ ⭐️

**Pull request merged**